web/packages/teleport/src/Instances/InstancesList.tsx

```diff
@@ -0,0 +1,352 @@
+/**
+ * Teleport
+ * Copyright (C) 2025 Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { Link } from 'react-router-dom';
+import styled from 'styled-components';
+
+import { Box, Flex, Text } from 'design';
+import { Danger } from 'design/Alert';
+import { ButtonBorder } from 'design/Button';
+import Table, { Cell } from 'design/DataTable';
+import * as Icons from 'design/Icon';
+import { Indicator } from 'design/Indicator';
+import { HoverTooltip } from 'design/Tooltip';
+import { CopyButton } from 'shared/components/CopyButton/CopyButton';
+import { useInfiniteScroll } from 'shared/hooks';
+
+import cfg from 'teleport/config';
+import { UnifiedInstance } from 'teleport/services/instances/types';
+
+type TableInstance = {
+  name: string;
+  version: string;
+  type: 'instance' | 'bot_instance';
+  original: UnifiedInstance;
+};
+
+export function InstancesList(props: {
+  data: UnifiedInstance[];
+  isLoading: boolean;
+  isFetchingNextPage: boolean;
+  error: Error | null;
+  hasNextPage: boolean;
+  sortField: string;
+  sortDir: string;
+  onSortChanged: (sortField: string, sortDir: string) => void;
+  onLoadNextPage: () => void;
+}) {
+  const {
+    data,
+    isLoading,
+    isFetchingNextPage,
+    error,
+    hasNextPage,
+    sortField,
+    sortDir,
+    onSortChanged,
+    onLoadNextPage,
+  } = props;
+
+  // Flatten the data
```

Comment:
Well technically this is more like unifying/normalizing it rather than flattening. ☝️🤓